app/controllers/logs_controller.rb

```diff
@@ -2,41 +2,46 @@ class LogsController < ApplicationController
   before_action :log_owner, only: [:index, :stockrooms]
 
   def index
-    @logs = Log.where(id: @user_logs).paginate(page: params[:page], per_page: 10)
+    # @logs = Log.where(id: @user_logs).paginate(page: params[:page], per_page: 10)
+    @logs = current_user.logs.paginate(page: params[:page], per_page: 10)
   end
 
   def stockrooms
     # Needs to update the route structure to get the stockroom ID:
     # @logs = Log.where(id: stockroom_logs(params[:stockroom_id])).paginate(page: params[:page], per_page: 10)
-    @logs = Log.where(id: stockroom_logs(15)).paginate(page: params[:page], per_page: 10)
+    @logs = Log.where(id: stockroom_log_ids(params[:id])).paginate(page: params[:page], per_page: 10)
   end
 
-  def stockroom_logs(stockroom_id)
-    @stockroom_logs = []
-    @logs = Log.all
-    @logs.each do |log|
-      if !log.supporting_details.stockroom.nil? && log.supporting_details.stockroom.id == stockroom_id
-        @stockroom_logs << log
-      end
-    end
-    @stockroom_logs
-  end
+  # def stockroom_logs(stockroom_id)
+  #   @stockroom_logs = []
+  #   @logs = Log.all
+  #   @logs.each do |log|
+  #     if !log.supporting_details.stockroom.nil? && log.supporting_details.stockroom.id == stockroom_id
+  #       @stockroom_logs << log
+  #     end
+  #   end
+  #   @stockroom_logs
+  # end
 
   private
 
-  def log_owner
-    @user_logs = []
-    @logs = Log.all
-    @logs.each do |log|
-      if log.supporting_details.user.id == current_user.id
-        @user_logs << log.id
-      # The two conditions below are in prevision of having several users per account
-      elsif !log.supporting_details.stockroom.nil? && current_user.stockroom_ids.include?(log.supporting_details.stockroom.id)
-        @user_logs << log.id
-      elsif !log.supporting_details.item.nil? && current_user.item_ids.include?(log.supporting_details.item.id)
-        @user_logs << log.id
-      end
-    end
-    @user_logs
+  def stockroom_log_ids(stockroom_id)
+    current_user.logs.map{|log| log.id if log.supporting_details.stockroom.id.to_s == stockroom_id}.compact
```

Comment:
So this the method of getting the log ids. 
It starts by looking the current_user logs (`current_user.logs`)
then loop inside using map method (`.map{}`)
 and in each log there will be a condition that will look on the supporting_details and find the stockroom id and compare if its match to the specified stockroom id(outside)  ( `if log.supporting_details.stockroom.id.to_s == stockroom_id)`
then there will be null values on the result. as the help of compact (`.compact`), it will remove the nil values.